doc/warm-reboot/NSF_Manager_HLD.md

```diff
@@ -0,0 +1,280 @@
+# NSF Manager HLD
```

Comment:
Could you prevent using the terminalogy "NSF" in this design doc? There is another active HLD relating gNOI API, and we are discussing use gNOI NSFT terminalogy to represent SONIC fast-reboot instead of SONiC warm-reboot, since there is another gNOI WARM already. ref: https://github.com/openconfig/gnoi/blob/98d6b81c6dfe3c5c400209f82a228cf3393ac923/system/system.proto#L128

May I propose "Warm Reboot Manager" in this HLD?

Reply:
Could you prevent using the terminology "NSF" in this design doc?

Reply:
I somehow prefer using gnoi WARM terminology over NSF. Would you mind change all the NSF->WARM in this HLD including PR title and markdown headers. I will not block this PR if using WARM in HLD.

Reply:
Replaced NSF with Warmboot. Updated the document tile, headings and diagrams.

Reply:
Thanks!